src/client/api.cpp

```diff
@@ -65,23 +65,53 @@ py::dict getRuleAttributes(const prt::RuleFileInfo* ruleFileInfo) {
 	auto ruleAttrs = py::dict();
 
 	for (size_t i = 0; i < ruleFileInfo->getNumAttributes(); i++) {
-		bool hidden = false;
 		const prt::RuleFileInfo::Entry* attr = ruleFileInfo->getAttribute(i);
-
+		bool hidden = false;
+		std::vector<std::vector<py::object>> annotations;
 		const std::wstring fullName(attr->getName());
 		if (fullName.find(L"Default$") != 0)
 			continue;
 		const std::wstring name = fullName.substr(8);
-		const prt::AnnotationArgumentType valueType = attr->getReturnType();
-		py::str type;
 
 		for (size_t f = 0; f < attr->getNumAnnotations(); f++) {
-			if (!(std::wcscmp(attr->getAnnotation(f)->getName(), ANNOT_HIDDEN))) {
+			const prt::Annotation* annot = attr->getAnnotation(f);
+			if (std::wcscmp(annot->getName(), ANNOT_HIDDEN) == 0) {
 				hidden = true;
 				break;
 			}
+
+			if (std::wcsncmp(annot->getName(), L"@", 1) == 0) {
+				std::vector<py::object> annotation;
+				annotation.push_back(py::cast(annot->getName())); //first reserve?
+
+				for (size_t u = 0; u < annot->getNumArguments(); u++) {
+					py::object annotationValue;
+					const prt::AnnotationArgument* annotationArg = annot->getArgument(u);
+					const prt::AnnotationArgumentType annotationValueType = annotationArg->getType();
+
+					if (annotationValueType == prt::AAT_STR)
+						annotationValue = py::cast(annotationArg->getStr());
+					else if (annotationValueType == prt::AAT_BOOL)
+						annotationValue = py::cast(annotationArg->getBool());
+					else if (annotationValueType == prt::AAT_FLOAT)
+						annotationValue = py::cast(annotationArg->getFloat());
+					else
+						annotationValue = py::cast("UNKNOWN_PARAMETER_VALUE_TYPE");
+
+					py::list annotationParameters;
+					annotationParameters.append(py::cast(annotationArg->getKey()));

```

Comment:
I wonder if we should introduce some python 'constant' NO_KEY to hide `#NULL#` key sentinel value